### PR TITLE
refactor(PVS/V1048): prevent "variable was assigned the same value" warning

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Comment by zeertzjq** [comment]
BTW, Vim also has this duplicate assignment.

**Comment by dundargoc** [comment]
Will send patch.

**Comment by zeertzjq** [comment]
<https://github.com/vim/vim/pull/9442>

File: src/nvim/memline.c [hunk 1]
```diff
@@ -1032,9 +1032,7 @@ void ml_recover(bool checkext)
   line_count = 0;
   idx = 0;              // start with first index in block 1
   error = 0;
-  buf->b_ml.ml_stack_top = 0;
```
Comment (bfredl) [inline src/nvim/memline.c:1035]: Hmm I think resetting `size` (and this index) is a good habit when clearing the corresponding pointer. what is the _other_ assignment this is duplicate to?

Comment (dundargoc) [reply src/nvim/memline.c:1035]:
```
buf->b_ml.ml_stack_top = 0;           // nothing in the stack
```

in line 854.

**Comment by dundargoc** [comment]
Need to merge https://github.com/neovim/neovim/pull/16857 before this.

File: src/nvim/memline.c (hunk 1 above)
Comment (bfredl) [reply src/nvim/memline.c:1035]: That's quite non-local. I would keep this for robustness (consider `ml_stack` an embedded struct with a coherent state).

Comment (dundargoc) [reply src/nvim/memline.c:1035]: Shall I add an ignore directive then?

**Pull request merged**